docs/retries.md

```diff
@@ -0,0 +1,267 @@
+```java
+RetrySettings.newBuilder()
+    .setInitialRetryDelay(Duration.ofMillis(200L))
+    .setRetryDelayMultiplier(2.0)
+    .setMaxRetryDelay(Duration.ofMillis(500L))
+    .setInitialRpcTimeout(Duration.ofMillis(1500L))
+    .setRpcTimeoutMultiplier(2.0)
+    .setMaxRpcTimeout(Duration.ofMillis(3000L))
+    .setTotalTimeout(Duration.ofMillis(10000L))
+    .build();
+```
+
+The following table shows the attempts:
+
+| Attempt Number 	| RPC Timeout 	| Retry Delay 	| Call Invoked 	| Call Ended 	|
+|----------------	|-------------	|-------------	|--------------	|------------	|
+| 1              	| 1500ms      	| 0ms         	| 0ms          	| 1500ms     	|
+| 2 (Retry)      	| 3000ms      	| 200ms       	| 1700ms       	| 4700ms     	|
+| 3 (Retry)      	| 4900ms      	| 400ms       	| 5100ms       	| 10000ms    	|
+
+The third retry’s RPC Timeout value is limited due the Total Timeout value. Using the multiplier (2.0) with the
+previous timeout value (3000ms) should result in an RPC Timeout of 6000ms. However, the RPC Timeout should not
+exceed the Total Timeout and is reduced to be the "time left" (10000 - 5100 = 4900).
+
+#### Example 3
+```java
+RetrySettings defaultRetrySettings =
+    RetrySettings.newBuilder()
+        .setInitialRetryDelay(Duration.ofMillis(200L))
+        .setRetryDelayMultiplier(2.0)
+        .setMaxRetryDelay(Duration.ofMillis(500L))
+        .setInitialRpcTimeout(Duration.ofMillis(500L))
+        .setRpcTimeoutMultiplier(2.0)
+        .setMaxRpcTimeout(Duration.ofMillis(2000L))
+        .setTotalTimeout(Duration.ofMillis(4000L))
+        .build();
+```
+
+The following table shows the attempts:
+
+| Attempt Number 	| RPC Timeout 	| Retry Delay 	| Call Invoked 	| Call Ended 	|
+|----------------	|-------------	|-------------	|--------------	|------------	|
+| 1              	| 500ms       	| 0ms         	| 0ms          	| 500ms      	|
+| 2 (Retry)      	| 1000ms      	| 200ms       	| 700ms        	| 1700ms     	|
+| 3 (Retry)      	| 1900ms      	| 400ms       	| 2100ms       	| 4000ms     	|
+
+Another example where the RPC Timeout is capped to not exceed the total timeout.
+
+## How to configure a custom retry configuration for an RPC
+1. Create the RetrySettings class with your custom configurations
+```java
+RetrySettings customRetrySettings =
+  RetrySettings.newBuilder()
+      // ... Retry Configurations
+      .build();
+```
+2. Create the StubSettings.Builder for your client and configure it for the RPC
+```java
+AssetServiceStubSettings.Builder assetStubSettingsBuilder = AssetServiceStubSettings.newBuilder();
+assetStubSettingsBuilder
+  .exportAssetsSettings()
+  // Set your custom Retry Settings
+  .setRetrySettings(customRetrySettings)
+  // Set your custom Retryable Codes
+  .setRetryableCodes(ImmutableSet.of(StatusCode.Code.DEADLINE_EXCEEDED));
+```
+
+The snippet above is using a mock service (i.e. EchoClient). The code snippet above is setting
```

Comment:
We can change this part to mention `AssetServiceClient` instead now.

Reply:
Ah, yep! Will update.